src/middleware/authenticateUser.js

```diff
@@ -0,0 +1,24 @@
+const User = require('../models/User')
+const { auth } = require('../constants/errors.json')
+const boom = require('boom')
+
+const authenticateUserAndAddId = (req) => {
+  return User.findOne({ username: req.user.username })
+    .then(user => {
+      if (!user) {
+        return Promise.reject(boom.unauthorized(auth.UNAUTHORIZED))
+      }
+
+      req.user.id = user.id
+      // user is returned for use in oauth
+      return req.user
+    })
+}
+
+module.exports = (req, res, next) => {
```

Comment:
Yeah, there is no real authentication happening in this file, all we do is check to see if there is a valid username on the request.user object.

I expect the logic overall is fine, but naming is problematic

Reply:
Yeah I mean we debated this name alot tbh. Isn't checking to see if a username is valid authenticating it? Verify instead of authenticate? What do you think of as authenticating?

Reply:
Checking the JWT / oauth token are valid